src/libs/PolicyUtils.ts

```diff
@@ -1087,6 +1087,29 @@ const sortWorkspacesBySelected = (workspace1: WorkspaceDetails, workspace2: Work
     return workspace1.name?.toLowerCase().localeCompare(workspace2.name?.toLowerCase() ?? '') ?? 0;
 };
 
+/**
+ * An eligible workspace is one that meets the following criteria:
+ * Submitters: workspaces where the submitter is a member of
+ * Approvers: workspaces where both the approver AND submitter are members of
+ * Admins: same as approvers OR workspaces where the admin is an admin of (note that the submitter is invited to the workspace in this case)
+ */
+const isWorkspaceEligibleForReportChange = (
+    newPolicy: OnyxEntry<Policy>,
+    reportOwnerAccountID: number | undefined,
+    reportManagerID: number | undefined,
+    currentUserLogin: string | undefined,
+): boolean => {
+    const curretUserAccountID = getCurrentUserAccountID();
+    if (curretUserAccountID === reportOwnerAccountID) {
+        return !!currentUserLogin && !!newPolicy?.employeeList?.[currentUserLogin];
+    }
+    if (curretUserAccountID === reportManagerID) {
+        const reportSubmitterLogin = (!!reportOwnerAccountID && getLoginsByAccountIDs([reportOwnerAccountID]).at(0)) ?? '';
```

Comment:
NAB but maybe it would be nice small improvement to add auxiliary function 
```typescript
function getLoginByAccountID(accountID: number): string | undefined {
    return allPersonalDetails?.[accountID]?.login;
}
```
and use it both here and it `getLoginsByAccountIDs` implementation?